tools/cldr-apps/js/src/esm/cldrDash.mjs

```diff
@@ -43,32 +52,107 @@ class DashData {
    * @param {Object} e (entry in old format, from json)
    */
   addEntry(cat, group, e) {
-    this.addCategory(cat);
-    this.catSize[cat]++;
-    if (!this.catFirst[cat]) {
-      this.catFirst[cat] = e.xpstrid;
+    try {
+      this.addCategory(cat);
+      this.catSize[cat]++;
+      if (!this.catFirst[cat]) {
+        this.catFirst[cat] = e.xpstrid;
+      }
+      if (CATS_ONE_PER_PAGE.includes(cat)) {
+        this.addCombinedEntry(cat, group, e);
+      } else if (this.pathIndex[e.xpstrid]) {
+        this.updateEntry(cat, group, e);
+      } else {
+        this.addNewEntry(cat, group, e);
+      }
+    } catch (err) {
+      console.error("Error in addEntry: " + err);
     }
-    if (this.pathIndex[e.xpstrid]) {
-      const dashEntry = this.pathIndex[e.xpstrid];
-      dashEntry.addCategory(cat);
-      dashEntry.setWinning(e.winning);
-      if (e.comment) {
-        dashEntry.setComment(e.comment);
+  }
+
+  addCombinedEntry(cat, group, e) {
+    try {
+      const page = group.page;
+      if (!this.pageCombinedEntries[cat]) {
+        this.pageCombinedEntries[cat] = {};
       }
-      if (e.subtype) {
-        dashEntry.setSubtype(e.subtype);
+      if (!this.pageCombinedEntries[cat][page]?.length) {
+        this.pageCombinedEntries[cat][page] = new Array(e.xpstrid);
+        this.addNewEntry(cat, group, e);
+      } else {
+        // TODO: make this work. unshift instead of push may be appropriate
+        // during an update (following a vote), since the combined notification
+        // is temporarily removed then added back, and in this case it may belong
+        // at the start of the array, not the end.
+        // Reference: https://unicode-org.atlassian.net/browse/CLDR-17658
+        if (this.updatingPath) {
```

Comment:
Bug: this.updatingPath is always false here, because here "this" = dashData (as intended), but below when this.updatingPath is set to true, "this" = cldrDash